pandas/tests/indexes/test_interval.py

```diff
@@ -491,6 +642,206 @@ def testcontains(self):
+        tm.assert_numpy_array_equal(Interval(1, 2, closed='left').overlaps(idx), np.array([0, 2]))
+        tm.assert_numpy_array_equal(Interval(0, 2, closed='left').overlaps(idx), np.array([0, 2]))
+        tm.assert_numpy_array_equal(Interval(3, 4, closed='left').overlaps(idx), np.array([3]))
+
+        tm.assert_numpy_array_equal(Interval(1, 3, closed='both').overlaps(idx), np.array([0, 1, 2]))
+        tm.assert_numpy_array_equal(Interval(1, 2, closed='both').overlaps(idx), np.array([0, 2]))
+        tm.assert_numpy_array_equal(Interval(0, 2, closed='both').overlaps(idx), np.array([0, 2]))
+        tm.assert_numpy_array_equal(Interval(3, 4, closed='both').overlaps(idx), np.array([3]))
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_intervalIndex_covers_interval_updated_behavior(self):
+
+        # class IntervalIndex:
+        #     def covers(self, other: Interval) -> IntegerArray1D
+
+        idx = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)], closed='right')
+
+        tm.assert_numpy_array_equal(idx.covers(Interval(1, 3, closed='right')), np.array([1, 2]))
+        tm.assert_numpy_array_equal(idx.covers(Interval(0, 3, closed='right')), np.array([0, 1, 2]))
+        tm.assert_numpy_array_equal(idx.covers(Interval(0, 2, closed='right')), np.array([0]))
+        tm.assert_numpy_array_equal(idx.covers(Interval(2, 4, closed='right')), np.array([1]))
+
+        tm.assert_numpy_array_equal(idx.covers(Interval(1, 3, closed='left')), np.array([]))
+        tm.assert_numpy_array_equal(idx.covers(Interval(0, 3, closed='left')), np.array([0]))
+        tm.assert_numpy_array_equal(idx.covers(Interval(0, 2, closed='left')), np.array([0]))
+        tm.assert_numpy_array_equal(idx.covers(Interval(2, 4, closed='left')), np.array([1]))
+
+        tm.assert_numpy_array_equal(idx.covers(Interval(1, 3, closed='both')), np.array([1, 2]))
+        tm.assert_numpy_array_equal(idx.covers(Interval(0, 5, closed='both')), np.array([0, 1, 2]))
+        tm.assert_numpy_array_equal(idx.covers(Interval(0, 2, closed='both')), np.array([0]))
+        tm.assert_numpy_array_equal(idx.covers(Interval(2, 4, closed='both')), np.array([1]))
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_intervalIndex_covers_intervalIndex(self):
+
+        # class IntervalIndex:
+        #     def covers(self, other: IntervalIndex) -> Tuple[IntegerArray1D, IntegerArray1D]
+
+        idx1 = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)], closed='right')
+        idx2 = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)], closed='left')
+        idx3 = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)], closed='both')
+
+        tm.assert_numpy_array_equal(idx.covers(idx1), (np.array([0,1,2,2]), np.array([0,1,1,2])))  # note: assert_numpy_array_equal is the wrong thing to call here, since we're testing for equality with a tuple of np arrays
```

Comment:
probably not :).

It would also be fine to cast the result to a numpy array, e.g., `np.array(idx.covers(idx1))`.